code/extensions/che-terminal/src/machine-exec-client.ts

```diff
@@ -136,6 +138,11 @@ export class MachineExecClient implements vscode.Disposable {
 	 * @returns a TerminalSession object to manage the created terminal session
 	 */
 	async createTerminalSession(component: string, commandLine?: string, workdir?: string, columns: number = 80, rows: number = 24): Promise<TerminalSession> {
+		if (commandLine) {
+			const bashrc = join(env.HOME!, '.bashrc');
+			commandLine = `test -f ${bashrc} >> /dev/null 2>&1 && source ${bashrc};${commandLine}`;
```

Comment:
It's worth creating a `.bashrc` that will cause an error (e.g. add `cat nonExistentFile.txt`) to see if that breaks running Devfile tasks with this approach. In theory, it shouldn't since we are using `;` instead of `&&`. However, I suspect any errors in calling `source ~/.bashrc` will be logged when running the Devfile task. I'm not sure if that is desired?

A potential advantage of https://github.com/che-incubator/che-code/pull/383 is that we only call `source ~/.bashrc` once, and [catch the error](https://github.com/che-incubator/che-code/pull/383/files#diff-9b705d84a45991e6f1435dfe87ccee1687896d9c179ede2b08c5a600878a421fR31-R36). 

In the current approach, if the call to `source ~/.bashrc` results in any errors, the errors will repeatedly be hit when running a devfile task.

Reply:
I like the idea to experiment with `.bashrc`. Let me then update the repository I'm using to test the pull request.

> However, I suspect any errors in calling source ~/.bashrc will be logged when running the Devfile task. I'm not sure if that is desired?

Good catch. I will experiment with this as well.